modules/server/src/main/scala/higherkindness/compendium/db/PgDBService.scala

```diff
@@ -22,18 +22,23 @@ import doobie.util.transactor.Transactor
 import doobie.implicits._
 import higherkindness.compendium.core.refinements.ProtocolId
 import higherkindness.compendium.db.queries.Queries
+import higherkindness.compendium.models.DBModels.MetaProtocolDB
+import higherkindness.compendium.models.IdlNames
 
 object PgDBService {
 
   def impl[F[_]: Async](xa: Transactor[F]): DBService[F] =
     new DBService[F] {
 
-      override def upsertProtocol(id: ProtocolId): F[Unit] =
-        Queries.upsertProtocolIdQ(id.value).run.void.transact(xa)
+      override def upsertProtocol(id: ProtocolId, idlName: IdlNames): F[Unit] =
+        Queries.upsertProtocolIdQ(id.value, idlName.entryName).run.void.transact(xa)
 
       override def existsProtocol(id: ProtocolId): F[Boolean] =
         Queries.checkIfExistsQ(id.value).unique.transact(xa)
 
+      override def selectProtocolById(id: ProtocolId): F[MetaProtocolDB] =
```

Comment:
Maybe we do not need to create `MetaProtocolDB`. This method just needs to retrieve the idl for the protocol. 

```suggestion
      override def getProtocolIdl(id: ProtocolId): F[IdlName] =
```

Reply:
I created `MetaProtocolDB` in order to have a model with all the metadata related to a `Protocol`. Maybe now it seems unneeded but, in the future, when we have to handle more metadata (version, owner, whatever...) I think that have this case class would make things easier. However, if you consider that we can get rid of it, we can delete